x-pack/plugins/apm/public/context/inspector/inspector_context.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { createContext, ReactNode, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import { RequestAdapter } from '../../../../../../src/plugins/inspector/common';
+import { InspectResponse } from '../../../typings/common';
+import { FetcherResult } from '../../hooks/use_fetcher';
+
+export interface InspectorContextValue {
+  addInspectorRequest: <Data>(result: FetcherResult<Data>) => void;
+  inspectorAdapters: { requests: RequestAdapter };
+}
+
+const value: InspectorContextValue = {
+  addInspectorRequest: () => {},
+  inspectorAdapters: { requests: new RequestAdapter() },
+};
+
+export const InspectorContext = createContext<InspectorContextValue>(value);
+
+export function InspectorContextProvider({
+  children,
+}: {
+  children: ReactNode;
+}) {
+  const history = useHistory();
+  const { inspectorAdapters } = value;
+
+  function addInspectorRequest(
+    result: FetcherResult<{
+      mainStatisticsData?: { _inspect?: InspectResponse };
+      _inspect?: InspectResponse;
+    }>
+  ) {
+    (
+      result.data?._inspect ??
+      result.data?.mainStatisticsData?._inspect ??
+      []
+    ).forEach((operation) => {
```

Comment:
nit: perhaps split this up to make it easier to read

```suggestion
	const operations = result.data?._inspect ?? result.data?.mainStatisticsData?._inspect ?? [];
    operations.forEach((operation) => {
```

Reply:
Fixed in b0da4dba8c2.